app/models/training_period.rb

```diff
@@ -101,4 +102,14 @@ def only_provider_led_mentor_training
       errors.add(:training_programme, 'Mentor training periods can only be provider-led')
     end
   end
+
+  def expression_of_interest_and_school_partnership_absent
+    if expression_of_interest.present?
+      errors.add(:expression_of_interest, 'must be absent for school-led training programmes')
+    end
+
+    if school_partnership.present?
+      errors.add(:school_partnership, 'must be absent for school-led training programmes')
```

Comment:
Won't this mean the error message we display in the error summary doesn't have the field prefix? I think this should be `'School partnership must be absent for school-led training programme'`.